lib/csvlint/validate.rb

```diff
@@ -93,6 +89,25 @@ def validate_metadata(io)
       build_info_messages(:assumed_header, :structure) if assumed_header
     end
 
+    def set_dialect(dialect)
+      begin
```

Comment:
I'd probably wrap this in `if @schema.tables.present?`, because this will only ever not error if we have a CSVW schema present, so there's an extra bit of work we don't need to do otherwise. Also, you could potentially add a schema error in the rescue block.